gapis/api/cmd_id_group.go

```diff
@@ -37,6 +37,7 @@ type CmdIDGroup struct {
 	Range    CmdIDRange // The range of commands this group (and items) represents.
 	Spans    Spans      // All sub-groups and sub-ranges of this group.
 	UserData interface{}
+	Parent   SubCmdIdx // SubCmdIdx of the SubCmdRoot if this CmdIDGroup belongs to a SubCmdRoot, otherwise just an empty SubCmdIdx
```

Comment:
Please can we discuss this CL before submitting. This code has now passed the threshold of complexity that means I'll never be able to understand it ever again.

Reply:
Sure. This is `Parent` is introduced in #1052 . Basically this is used in `Index()` when the CmdIDGroup is a group of subcommands.

Reply:
I think another option is to create a new SubCmdIDGroup to group the subcommands, but then we will need to extend the AddGroup() to the new type